addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -342,6 +342,38 @@ def _import_retrieve_and_fill_partner(self, invoice, name, phone, mail, vat, cou
             if vat and self.env['res.partner']._run_vat_test(vat, country, invoice.partner_id.is_company):
                 invoice.partner_id.vat = vat
 
+    def _import_retrieve_and_fill_partner_bank_details(self, invoice, bank_details):
+        """ Retrieve the bank account, if no matching bank account is found, create it
+        """
+        if invoice.journal_id.type == 'purchase':
+            partner = invoice.partner_id
+        elif invoice.journal_id.type == 'sale':
+            partner = self.env.company.partner_id
+        else:
+            return
+
+        banks_to_create = []
+
+        for account_number in bank_details:
+            if account_number:
+                partner_bank = self.env['res.partner.bank'].search(
+                    [('company_id', 'in', [False, self.env.company.id]), ('acc_number', '=', account_number)],
+                    limit=1
+                )
+
+                if partner_bank:
+                    if partner_bank.partner_id.id == partner.id:
+                        invoice.partner_bank_id = partner_bank
+                        return
```

Comment:
the logic is a little weird...
if there are several account numbers, but the last one exists, the other won't be created. Is that what we want?